hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieMergedLogRecordScanner.java

```diff
@@ -82,6 +88,9 @@ public class HoodieMergedLogRecordScanner extends AbstractHoodieLogRecordReader
   public final HoodieTimer timer = HoodieTimer.create();
   // Map of compacted/merged records
   private final ExternalSpillableMap<String, HoodieRecord> records;
+
+  private final ExternalSpillableMap<String, HashMap<String, HoodieRecord>> nonUniqueKeyRecords;
```

Comment:
serializing the hashmap to spillable disk - may not be efficient? or is that okay? I wonder if we need another holder object.

Reply:
Guava and Chronicle supports persistent multi-map structures (just from the documentation of the library)